Polish Writing Basic Smart Contract page (#969)
casper-network/docs: source/docs/casper/developers/writing-onchain-code/simple-contract.md

    └── Cargo.toml
```

> The project structure will be different while designing the full stack architecture. This will be expanded upon while describing the dApps.

Reviewer: Please comment out this line until the new page is created. This is a simple example just to get someone started. For a more thorough example, a link can be added here when the guide is available.

Author: @jonas089 This still has to be addressed.

Reviewer: Ideally, we only include this line once the guide is ready, but if you have an issue number, please place it here, and we can resolve the comment.

### Creating the Project Manually

:::tip

> As a beginner it is not advised to start with the manual project structure.
> Structure created automatically with `cargo casper` contains everything that is needed to start coding.

Reviewer: Actually, this section is for developers who want to set things up from scratch; otherwise, we wouldn't include it.
I would not advise against this section, but we could move it after the  "Creating the Project Automatically" section.
```suggestion
If you are a beginner, [creating the structure automatically](#creating-the-project-automatically) with `cargo casper` contains everything you need to start coding.
```

Reviewer: Thanks. The commit suggestion wasn't applied, though. I will apply it.

    The [Testing Smart Contracts](./testing-contracts.md) guide explains how to update the tests using example code.

## Writing a Basic Smart Contract

This section covers the process of writing a smart contract in Rust, using example code from the [counter contract](https://github.com/casper-ecosystem/counter/). This simple contract allows callers to increment and retrieve an integer. Casper provides a [contract API](https://docs.rs/casper-contract/latest/casper_contract/contract_api/index.html) within the [`casper_contract`](https://docs.rs/casper-contract/latest/casper_contract/index.html) crate.

:::info

Important syntax elements used frequently in Rust:

- [Match](https://doc.rust-lang.org/rust-by-example/flow_control/match.html)<br />
- [Array](https://doc.rust-lang.org/rust-by-example/primitives/array.html)<br />
- [Loop](https://doc.rust-lang.org/rust-by-example/flow_control/loop.html)<br />
- [Vectors](https://doc.rust-lang.org/rust-by-example/std/vec.html)<br />
- [Functions](https://doc.rust-lang.org/rust-by-example/fn.html)<br />

To be able to comfortably write code in Rust it is crucial to understand these topics before going further into the examples.

:::

> 

Reviewer: This page assumes that readers have a basic understanding of Rust. I recommend moving this section to a new guide or page that could be "An introduction to Rust", "Important Rust Syntax", or "Tips for Rust Contracts", under Resources.
It would also be good to include references to the Rust Book or Rust by Example. However, that is another topic that would be better placed as a resource.

Author: @KMCreatesWorlds This still has to be addressed.

Reviewer: Ok. If you have an issue number, please place it here, and we can resolve the comment so that this request is captured.
